Add support for Cartesia synthesizer

rjheeta opened the pull request:
This PR is to add Cartesia (https://cartesia.ai/) as a new synthesizer for TTS. Please review and let me know if you have any suggestions.

I'd need your assistance to double check the `generate` call in  `cartesia_synthesizer.py` to make sure we're doing this in the most efficient manner.

ajar98 reviewed (commented):
this is awesome! i'd like to try to use `stream=True` - let me get back to you on support for that

rjheeta commented:
@ajar98 Sounds good. Interesting you mention that because I noticed there was a bit of a lag in the audio -- and I suspect (and hope) that using `stream=True` can fix this. I just wasn't sure how to incorporate that with the existing Synthesizer classes to allow the audio to continually stream versus waiting for it to return.

rjheeta commented:
For your review. This is with setting `stream=True`. This works, but I don't know if it's optimal or if I've missed something. Either way, could use a second set of eyes when you get some time. I'm also not clear if the fixed `sample_rate` of 44100 is the right idea (although this seems to be the same each time)

```
    async def create_speech(
        self,
        message: BaseMessage,
        chunk_size: int,
        bot_sentiment: Optional[BotSentiment] = None,
    ) -> SynthesisResult:
        create_speech_span = tracer.start_span(
            f"synthesizer.{SynthesizerType.CARTESIA.value.split('_', 1)[-1]}.create_total",
        )
        generator = await self.client.generate(
            transcript=message.text,
            voice=self.voice_embedding,
            stream=True,
            model_id=self.model_id,
            data_rtype='bytes',
            output_format='pcm'
        )
        create_speech_span.end()
        convert_span = tracer.start_span(
            f"synthesizer.{SynthesizerType.CARTESIA.value.split('_', 1)[-1]}.convert",
        )
        
        sample_rate = 44100
        audio_file = io.BytesIO()

        with wave.open(audio_file, 'wb') as wav_file:
            wav_file.setnchannels(1)
            wav_file.setsampwidth(2)
            wav_file.setframerate(sample_rate)
            async for chunk in generator:
                raw_data = chunk['audio']
                wav_file.writeframes(raw_data)
        audio_file.seek(0)

        result = self.create_synthesis_result_from_wav(
            synthesizer_config=self.synthesizer_config,
            file=audio_file,
            message=message,
            chunk_size=chunk_size,
        )
        convert_span.end()
        return result
```

rjheeta commented:
Also thoughts on using `experimental_mp3_streaming_output_generator` to further reduce audio latency? 

https://github.com/vocodedev/vocode-python/blob/9b24020439d395f54c9a89bb60b8833f68d918e9/vocode/streaming/synthesizer/base_synthesizer.py#L245

ajar98 commented:
subsumed in current release!!
